modules/gdscript/gdscript_parser.cpp

```diff
@@ -3739,6 +3747,15 @@ void GDScriptParser::_parse_class(ClassNode *p_class) {
 						arguments.push_back(argname);
 #ifdef DEBUG_ENABLED
 						arguments_usage.push_back(0);
+
+						// Try to find variable name shadowing checking from the actual class
+						for (ClassNode *class_iterator = p_class; class_iterator; class_iterator = class_iterator->owner) {
```

Comment:
Again, no need for iterator and the check should be in `_check_function_types()`.